src/cryptoadvance/specter/wallet.py

```diff
@@ -436,27 +438,47 @@ def _rescan_utxo_thread(self, explorer=None):
         ])
         # handle missing transactions now
         # if Tor is running, requests will be sent over Tor
+        tor_explorers = [
+            'http://explorerzydxu5ecjrkwceayqybizmpjjznk5izmitf2modhcusuqlid.onion',
+            'http://mempoolhqx4isw62xs7abwphsq7ldayuidyx2v2oethdhhj6mlo2r6ad.onion'
+        ]
+        if self.rpc.getblockchaininfo()['chain'] != 'main':
+            tor_explorers = [tor_explorer + '/testnet' for tor_explorer in tor_explorers]
+        explorers = []
         if explorer is not None:
             try:
                 requests_session = requests.Session()
                 requests_session.proxies['http'] = 'socks5h://localhost:9050'
                 requests_session.proxies['https'] = 'socks5h://localhost:9050'
-                requests_session.get(explorer)
             except Exception:
                 requests_session = requests.Session()
+            for tor_explorer in tor_explorers:
+                try:
+                    requests_session.get(tor_explorer)
+                    explorers.append(tor_explorer)
+                except Exception:
+                    pass
             # make sure there is no trailing /
             explorer = explorer.rstrip("/")
             try:
                 # get raw transactions
-                raws = [requests_session.get(
-                            f"{explorer}/api/tx/{tx['txid']}/hex"
-                            ).text
-                        for tx in missing]
+                raws = []
+                proofs = []
                 # get proofs
-                proofs = [requests_session.get(
-                            f"{explorer}/api/tx/{tx['txid']}/merkleblock-proof"
-                            ).text
-                          for tx in missing]
+                for tx in missing:
+                    if requests_session.proxies.get('http', None) == 'socks5h://localhost:9050':
+                        try:
+                            renew_tor_ip()
```

Comment:
after renew of a TOR ip we need to create new session. The old one will use the old IP address.

Reply:
Fixed with 6a900d95e9f77975a1ed826a77dd079a1a29c9f2